### ISSUE TITLE
Upgrade Backend python module to 3.9

### ISSUE THREAD

**Issue by manasaV3** [chanzuckerberg/napari-hub#1153]
As Python 3.8 is nearing EOL and will shortly stop getting security updates, we should upgrade the backend module from Python 3.8  to python 3.9.

**Comment by richaagarwal** [comment]
I think this is a duplicate of https://github.com/chanzuckerberg/napari-hub/issues/1043, but we can close the older one.  

Would it make more sense to upgrade all modules to the latest stable version, which is 3.11? That would make sense from a longer-term maintenance point of view so that we don't need to revisit this for several years.

**Comment by manasaV3** [comment]
Thanks for catching the duplicate ticket. :) 

I had opted for 3.9 as the gap between 3.8 and 3.9 was small and would be able to buy us more time. But I agree that migrating it to the latest stable version would be better, and having a parent ticket for that would make sense. :) 

Also, I created this and the plugins as two different issues because the effort needed for the plugins module was more straightforward for migration to 3.9 compared to the backend.
